src/reducers/index.ts

```diff
@@ -52,21 +52,50 @@ function checkStoreInvariants(state: StoreState) {
     state.currentMatchIndex >= -1 &&
       state.currentMatchIndex <= state.matchesList.length
   );
+
+  checkCondition(
+    'every matchId in matchesList is also present in matchIdToMatchState',
```

Comment:
We actually need the other direction :)
Match state will be loaded lazily.


  // Match state will be loaded lazily, i.e.,
  // not every match in matchesList will have an entry in matchIdToMatchState.
  matchIdToMatchState: MatchIdToMatchState;